pkg/plugin/bundle.go

```diff
@@ -31,19 +31,52 @@ type bundle struct {
 	deprecateWarning         string
 }
 
+type BundleOption func(*bundle)
+
+func WithName(name string) BundleOption {
+	return func(opts *bundle) {
+		opts.name = name
+	}
+}
+
+func WithVersion(version Version) BundleOption {
+	return func(opts *bundle) {
+		opts.version = version
+	}
+}
+
+func WithPlugins(plugins ...Plugin) BundleOption {
+	return func(opts *bundle) {
+		opts.plugins = plugins
+	}
+}
+
+func WithDeprecationMessage(msg string) BundleOption {
+	return func(opts *bundle) {
+		opts.deprecateWarning = msg
+	}
+
+}
+
 // NewBundle creates a new Bundle with the provided name and version, and that wraps the provided plugins.
 // The list of supported project versions is computed from the provided plugins.
-func NewBundle(name string, version Version, deprecateWarning string, plugins ...Plugin) (Bundle, error) {
-	supportedProjectVersions := CommonSupportedProjectVersions(plugins...)
+func NewBundle(opts ...BundleOption) (Bundle, error) {
```

Comment:
Could we keep here the old func as it is and add a new func using the Options so that we do not introduce a breaking change? See that the PR is fafling in the API diff check.

Reply:
Should revert the changes in the docs too, right?